frontend/src/api/appwriteApi.js

```diff
@@ -201,6 +195,45 @@ let api = {
 		});
 	},
 
+	/**
+	 * Storage API
+	 **/
+	imageToFileID(imageID) {
+		return imageID && `img${imageID}`;
+	},
+
+	async loadImageFromDatabase(imageID) {
+		const fileID = this.imageToFileID(imageID);
+		try {
+			return await this.sdk.storage.getFile(fileID);
+		} catch(e) {
+			console.error(`Couldn't load image with ID ${fileID} from database! Likely does not exist or was deleted.`);
+			return null;
+		}
+	},
+
+	async saveImageToDatabase(image, imageID) {
+		const fileID = this.imageToFileID(imageID);
+		
+		this.sdk.storage.deleteFile(fileID)
+			.catch(() => console.log(`No image to update. Create new image ${fileID}!`));
+
+		try {
+			const readPermissions = ["*"];
+			console.debug("save file with ID", fileID);
+			await this.sdk.storage.createFile(fileID, image, readPermissions);
+			return true;
+		} catch(e) {
+			console.error(`Cannot save image with ID ${fileID}, likely no write permissions.`);
+			return false;
+		}
+	},
+
+	removeImageFromDatabase(imageID) {
+		const fileID = this.imageToFileID(imageID);
+		return this.sdk.storage.deleteFile(fileID)
```

Comment:
Good remark!
I indeed removed the writePermissions from the file creation call when I made the PR.

Announcements can only be created/edited by admins at the moment (using `AdminArea` component) so that deleting announcements (or updating images) will also delete the old picture. I had "bad request" problems that's why I removed the writePermissions on file creation. I used the same way like Que to set the write Permissions. Maybe, it wasn't the problem.

Reply:
The write permission should be set to `["team:{adminTeamID}"]` for sure.

Reply:
Also, just because an area is protected by the `AdminArea` component in the frontend, you cannot expect that the user finds a way to access that area anyways (e.g. if someone knows how to read through our code in the browser which isn't too difficult). That is why any action in the admin area always also is protected by admin permissions through the backend.

Reply:
Good point. I didn't exactly know, where which code is executed. Is it right that the functions are executed on server and the JSX is processed by the client? Or is everything run on client side?

In the most recent version in my other PR, I changed some things back due to the API overhead created by the AdminArea component. It rather makes sense for less-frequently updated content.

And if an attacker could potentially remove the AdminArea on client side, would these API calls have effect when used? Or rather not, because of lacking Admin rights?